pallene/checker.lua

```diff
@@ -797,7 +799,7 @@ function Checker:check_exp_synthesize(exp)
             exp._type = types.T.Integer()
 
         else
-            error("impossible")
+            typedecl.tag_error(tag, string.format("unknown binary operator '%s'.", op))
```

Comment:
Again, the error message should reference `op`, not the `tag`. With the error message I suggested I think we could use `typedecl.tag_error(op)`.